packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -80,18 +80,19 @@ const useDeleteWithConfirmController = ({
     const [deleteOne, { loading }] = useDelete(resource, null, null, {
         action: CRUD_DELETE,
         onSuccess: () => {
+            setOpen(false);
```

Comment:
A suggestion: It could be the occasion to add an `onEnd` event that would fire on both success and failure.
But yet again maybe it is overkill.